crates/recursor/src/recursor.rs

```diff
@@ -28,6 +28,11 @@ use crate::{
 pub struct RecursorBuilder {
     ns_cache_size: usize,
     record_cache_size: usize,
+    /// This controls how many nested lookups will be attempted to resolve a CNAME chain. It
+    /// defaults to 12 to support the longest real-world CNAME chains the authors could find,
+    /// but is user-configurable to support longer or shorter limits.  Setting it to 0 will
+    /// disable the recursion limit check, and is not recommended.
+    recursion_limit: u8,
```

Comment:
Nit: I think we should describe what it does here, but describe how you arrived at the default value in the `Default` impl, where we actually set the 12. Also I think it would be useful to add the link to the unbound issue you referenced.